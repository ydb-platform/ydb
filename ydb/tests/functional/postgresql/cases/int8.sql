--
-- INT8
-- Test int8 64-bit integers.
--
CREATE TABLE INT8_TBL(q1 int8, q2 int8, primary key (q1, q2));
INSERT INTO INT8_TBL (q1, q2) VALUES('  123   ','  456');
INSERT INTO INT8_TBL (q1, q2) VALUES('123   ','4567890123456789');
INSERT INTO INT8_TBL (q1, q2) VALUES('4567890123456789','123');
INSERT INTO INT8_TBL (q1, q2) VALUES(+4567890123456789,'4567890123456789');
INSERT INTO INT8_TBL (q1, q2) VALUES('+4567890123456789','-4567890123456789');
-- bad inputs
INSERT INTO INT8_TBL(q1) VALUES ('      ');
INSERT INTO INT8_TBL(q1) VALUES ('xxx');
INSERT INTO INT8_TBL(q1) VALUES ('3908203590239580293850293850329485');
INSERT INTO INT8_TBL(q1) VALUES ('-1204982019841029840928340329840934');
INSERT INTO INT8_TBL(q1) VALUES ('- 123');
INSERT INTO INT8_TBL(q1) VALUES ('  345     5');
INSERT INTO INT8_TBL(q1) VALUES ('');
SELECT * FROM INT8_TBL;
-- int8/int8 cmp
SELECT * FROM INT8_TBL WHERE q2 = 4567890123456789;
SELECT * FROM INT8_TBL WHERE q2 <> 4567890123456789;
SELECT * FROM INT8_TBL WHERE q2 < 4567890123456789;
SELECT * FROM INT8_TBL WHERE q2 > 4567890123456789;
SELECT * FROM INT8_TBL WHERE q2 <= 4567890123456789;
SELECT * FROM INT8_TBL WHERE q2 >= 4567890123456789;
-- int8/int4 cmp
SELECT * FROM INT8_TBL WHERE q2 = 456;
SELECT * FROM INT8_TBL WHERE q2 <> 456;
SELECT * FROM INT8_TBL WHERE q2 < 456;
SELECT * FROM INT8_TBL WHERE q2 > 456;
SELECT * FROM INT8_TBL WHERE q2 <= 456;
SELECT * FROM INT8_TBL WHERE q2 >= 456;
