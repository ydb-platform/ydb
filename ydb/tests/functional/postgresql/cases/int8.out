--
-- INT8
-- Test int8 64-bit integers.
--
CREATE TABLE INT8_TBL(q1 int8, q2 int8, primary key (q1, q2));
INSERT INTO INT8_TBL (q1, q2) VALUES('  123   ','  456');
INSERT INTO INT8_TBL (q1, q2) VALUES('123   ','4567890123456789');
INSERT INTO INT8_TBL (q1, q2) VALUES('4567890123456789','123');
INSERT INTO INT8_TBL (q1, q2) VALUES(+4567890123456789,'4567890123456789');
INSERT INTO INT8_TBL (q1, q2) VALUES('+4567890123456789','-4567890123456789');
-- bad inputs
INSERT INTO INT8_TBL(q1) VALUES ('      ');
ERROR:  invalid input syntax for type bigint: "      "
LINE 1: INSERT INTO INT8_TBL(q1) VALUES ('      ');
                                         ^
INSERT INTO INT8_TBL(q1) VALUES ('xxx');
ERROR:  invalid input syntax for type bigint: "xxx"
LINE 1: INSERT INTO INT8_TBL(q1) VALUES ('xxx');
                                         ^
INSERT INTO INT8_TBL(q1) VALUES ('3908203590239580293850293850329485');
ERROR:  value "3908203590239580293850293850329485" is out of range for type bigint
LINE 1: INSERT INTO INT8_TBL(q1) VALUES ('39082035902395802938502938...
                                         ^
INSERT INTO INT8_TBL(q1) VALUES ('-1204982019841029840928340329840934');
ERROR:  value "-1204982019841029840928340329840934" is out of range for type bigint
LINE 1: INSERT INTO INT8_TBL(q1) VALUES ('-1204982019841029840928340...
                                         ^
INSERT INTO INT8_TBL(q1) VALUES ('- 123');
ERROR:  invalid input syntax for type bigint: "- 123"
LINE 1: INSERT INTO INT8_TBL(q1) VALUES ('- 123');
                                         ^
INSERT INTO INT8_TBL(q1) VALUES ('  345     5');
ERROR:  invalid input syntax for type bigint: "  345     5"
LINE 1: INSERT INTO INT8_TBL(q1) VALUES ('  345     5');
                                         ^
INSERT INTO INT8_TBL(q1) VALUES ('');
ERROR:  invalid input syntax for type bigint: ""
LINE 1: INSERT INTO INT8_TBL(q1) VALUES ('');
                                         ^
SELECT * FROM INT8_TBL;
        q1        |        q2         
------------------+-------------------
              123 |               456
              123 |  4567890123456789
 4567890123456789 | -4567890123456789
 4567890123456789 |               123
 4567890123456789 |  4567890123456789
(5 rows)

-- int8/int8 cmp
SELECT * FROM INT8_TBL WHERE q2 = 4567890123456789;
        q1        |        q2        
------------------+------------------
              123 | 4567890123456789
 4567890123456789 | 4567890123456789
(2 rows)

SELECT * FROM INT8_TBL WHERE q2 <> 4567890123456789;
        q1        |        q2         
------------------+-------------------
              123 |               456
 4567890123456789 | -4567890123456789
 4567890123456789 |               123
(3 rows)

SELECT * FROM INT8_TBL WHERE q2 < 4567890123456789;
        q1        |        q2         
------------------+-------------------
              123 |               456
 4567890123456789 | -4567890123456789
 4567890123456789 |               123
(3 rows)

SELECT * FROM INT8_TBL WHERE q2 > 4567890123456789;
 q1 | q2 
----+----
(0 rows)

SELECT * FROM INT8_TBL WHERE q2 <= 4567890123456789;
        q1        |        q2         
------------------+-------------------
              123 |               456
              123 |  4567890123456789
 4567890123456789 | -4567890123456789
 4567890123456789 |               123
 4567890123456789 |  4567890123456789
(5 rows)

SELECT * FROM INT8_TBL WHERE q2 >= 4567890123456789;
        q1        |        q2        
------------------+------------------
              123 | 4567890123456789
 4567890123456789 | 4567890123456789
(2 rows)

-- int8/int4 cmp
SELECT * FROM INT8_TBL WHERE q2 = 456;
 q1  | q2  
-----+-----
 123 | 456
(1 row)

SELECT * FROM INT8_TBL WHERE q2 <> 456;
        q1        |        q2         
------------------+-------------------
              123 |  4567890123456789
 4567890123456789 | -4567890123456789
 4567890123456789 |               123
 4567890123456789 |  4567890123456789
(4 rows)

SELECT * FROM INT8_TBL WHERE q2 < 456;
        q1        |        q2         
------------------+-------------------
 4567890123456789 | -4567890123456789
 4567890123456789 |               123
(2 rows)

SELECT * FROM INT8_TBL WHERE q2 > 456;
        q1        |        q2        
------------------+------------------
              123 | 4567890123456789
 4567890123456789 | 4567890123456789
(2 rows)

SELECT * FROM INT8_TBL WHERE q2 <= 456;
        q1        |        q2         
------------------+-------------------
              123 |               456
 4567890123456789 | -4567890123456789
 4567890123456789 |               123
(3 rows)

SELECT * FROM INT8_TBL WHERE q2 >= 456;
        q1        |        q2        
------------------+------------------
              123 |              456
              123 | 4567890123456789
 4567890123456789 | 4567890123456789
(3 rows)
